int16 sender
int16 recipient
string[] dims
int16 matrixDim
float64[] infMat
float64[] infVec